openwisp_users/api/filters.py

```diff
@@ -0,0 +1,159 @@
+import swapper
+from django.core.exceptions import ValidationError
+from rest_framework.exceptions import NotFound
+
+Organization = swapper.load_model('openwisp_users', 'Organization')
+
+
+class FilterByOrganization:
+    """
+    Filter queryset based on the access to the organization
+    of the associated model. Use on of the sub-classes
+    """
+
+    @property
+    def _user_attr(self):
+        raise NotImplementedError()
+
+    def get_queryset(self):
+        qs = super().get_queryset()
+        if self.request.user.is_superuser:
+            return qs
+        return self.get_organization_queryset(qs)
+
+    def get_organization_queryset(self, qs):
+        return qs.filter(organization__in=getattr(self.request.user, self._user_attr))
```

Comment:
There's a little issue here, some models do not have the `organization` relation on their model but on a parent.
In those cases this won't work.

For that reason, I suggest adding another class attribute called `organization_lookup`, which defaults to `organization__in`.

```python
class FilterByOrganization:
    # ...
    organization_lookup = 'organization__in'

    def get_organization_queryset(self, qs):
        return qs.filter(**{
            self.organization_lookup: getattr(self.request.user, self._user_attr)
        })
```